api/v1alpha1/clusterconfig_types.go

```diff
@@ -321,6 +323,9 @@ type Addons struct {
 
 	// +optional
 	NFD *NFD `json:"nfd,omitempty"`
+
+	// +optional
+	CSIProviders CSIProviders `json:"csi,omitempty"`
```

Comment:
I see removed the pointer, but if its optional the convention is for it to be a pointer.
```suggestion
	CSIProviders *CSIProviders `json:"csi,omitempty"`
```